linera-client/src/persistent/indexed_db.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ops::Deref;
+
+use indexed_db_futures::prelude::*;
+use wasm_bindgen::JsValue;
+use wasm_bindgen_futures::wasm_bindgen;
+use web_sys::DomException;
+
+use super::LocalPersist;
+
+pub struct IndexedDb<T> {
```

Comment:
nit: I would suggest (again) a more concrete name like `IndexedDbValue` here, in line with the `Deref` below. I know that `Arc` is not named `AtomicallyReferenceCountedValue` but we're not writing the core Rust library here :)

Reply:
I'm not sure how this makes it more concrete — anything you can write an expression for is a value! I think in the context of Rust the word ‘value’ is used in names for some kind of dynamically-typed structure that you can't say anything more about, e.g. `serde_json::Value`.

Reply:
"value" is the name that you chose for the field holding the main piece of data. A step in the right direction would be to add a comment explaining what this container is doing.

Reply:
At the value level yes — but it's an information-free name.  If not for the other fields I would have left this as a tuple struct.

I'll add a doc comment!